test/ably/resthttp_test.py

```diff
@@ -135,3 +135,28 @@ def test_custom_http_timeouts(self):
         self.assertEqual(ably.http.http_open_timeout, 8)
         self.assertEqual(ably.http.http_max_retry_count, 6)
         self.assertEqual(ably.http.http_max_retry_duration, 20)
+
+    def test_no_host_fallback_if_environment_is_set(self):
+        # RSC15b
+        ably = AblyRest(token="foo", environment="sandbox")
+        self.assertEqual(ably.options.fallback_hosts, [])
+
+    def test_no_host_fallback_if_custom_host_is_set(self):
+        # RSC15b
+        ably = AblyRest(token="foo", rest_host="example.org")
+        self.assertEqual(ably.options.fallback_hosts, [])
+
+    def test_no_host_fallback_if_use_default_is_false_without_custom_list(self):
+        # RSC15b
+        ably = AblyRest(token="foo", fallback_hosts_use_default=False)
+        self.assertEqual(ably.options.fallback_hosts, [])
```

Comment:
There is nowhere in the spec that states when `fallback_hosts_use_default` is false it does not try, there is only a spec to state that when it is true and it is using a custom host that it will then try the fallback hosts as opposed to the default behaviour with custom hosts where it won't.